### ISSUE TITLE
"SQL worked" messages on the console

### ISSUE THREAD

**Issue by lassoan** [commontk/CTK#1186]
Lots of "SQL worked" messages are logged in the console when the application is built in debug mode.

I don't think that successful operations need to be logged, especially with this frequency. Setting to TRACE level might be considered, but the performance hit may be significant. So, the best would be to remove these logs or be able to optionally include it with #ifdef/#endif.

**Comment by Punzo** [comment]
for the moment I have set those logging at trace level in https://github.com/commontk/CTK/pull/1184.
NOTE ctk levels:
1) debug mode: debug
2) release mode: warn 

setting those logging to trace, clean 90% of the logging in debug mode. Note that if the level is `debug` and a static logger calls `logger.trace`, the additional computations are only an if condition https://github.com/commontk/CTK/blob/master/Libs/Core/ctkLogger.cpp#L72-L80

**Comment by Punzo** [comment]
> or example, `"SQL worked!\n SQL: " + query.lastQuery()` and similar string operations can be expensive. Since database operations are expensive as well, the overhead is probably acceptable.

uhm that's true, you are right. I will add #ifdef/#endif.

**Comment by jcfr** [comment]
>  I will add #ifdef/#endif.

I think having the ability to toggle the level at runtime makes sense.

Unless you were talking about excluding some code based on the debug/release, I am not sure we will want to add `#ifdef/#endif` associated a "trace" CMake option

**Comment by Punzo** [comment]
@lassoan  regarding (3), we could use #ifndef QT_NO_DEBUG as suggested by @jcfr. See https://github.com/commontk/CTK/pull/1184/commits/94eee0f61f3771faa1f6f8136756a912a2a612c8

**Comment by lassoan** [comment]
I agree that log level should be adjustable at runtime. But trace level logs are a special case. Many times you see trace level logging code commented out. This is the case in CTK, too: in the single class where trace-level log is used (in `ctkVTKAbstractView.cpp`), the lines are commented out. This is because usually you need these very low level logs extremely rarely (probably only once, when you debugged some complex issues) and then you don't want to slow down the application with them. Traces can be inexpensive if they are implemented with a macro, but by using `logger.trace()` the log message is always created, regardless of the current log level.

> we could use #ifndef QT_NO_DEBUG as suggested by @jcfr. 

Behavior changes depending on debug/non-debug mode should be avoided. Any extra calls or missed calls in devbug mode can change the application behavior so that a bug in release mode cannot be reproduced in debug mode. So, unless there is no other way of solving some issue, I would not add QT_NO_DEBUG-dependent behavior.

I would prefer a simple solution, such as:
- A. deleting the trace level log code
- B. adding a local `#define` (and `#ifdef/#endif` around trace level log code) in the .cpp file that developers can enable when troubleshooting some tough issues
- C. add a macro for trace level logging (that would avoid putting constructing the log message if there is no trace level log)
- D. comment out the trace level log code

My order of preference is A = most preferred, D = least preferred, but all acceptable.

It would be better to do this in a separate PR from #1184 because that PR is too big and contains another mostly unrelated feature.

**Comment by Punzo** [comment]
@lassoan ok I opened a PR for (A) https://github.com/commontk/CTK/pull/1187. Please have a look if you want to "save" some of the logging that I have removed (I removed all the ones that makes the terminal very dirty).
